components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java

```diff
@@ -801,18 +816,34 @@ private static void createAuditMessage(String action, String target, JSONObject
      * @param result              Result value
      */
     private void auditAccountLock(String action, String target, String userStoreDomainName, boolean isAdminInitiated,
-                                  String errorMsg, String result) {
+                                  String errorMsg, String result, Boolean isModifiedStatus) {
 
         JSONObject dataObject = new JSONObject();
         dataObject.put(AuditConstants.REMOTE_ADDRESS_KEY, MDC.get(AuditConstants.REMOTE_ADDRESS_QUERY_KEY));
         dataObject.put(AuditConstants.USER_AGENT_KEY, MDC.get(AuditConstants.USER_AGENT_QUERY_KEY));
         dataObject.put(AuditConstants.SERVICE_PROVIDER_KEY, MDC.get(AuditConstants.SERVICE_PROVIDER_QUERY_KEY));
         dataObject.put(AccountConstants.ADMIN_INITIATED, isAdminInitiated);
         dataObject.put(AuditConstants.USER_STORE_DOMAIN, userStoreDomainName);
+        dataObject.put(AuditConstants.IS_MODIFIED_STATUS, isModifiedStatus);
 
         if (AuditConstants.AUDIT_FAILED.equals(result)) {
             dataObject.put(AuditConstants.ERROR_MESSAGE_KEY, errorMsg);
         }
         createAuditMessage(action, target, dataObject, result);
     }
+
+    private void publishPreAccountLockedEvent(String accountLockedEventName, Map<String, Object> map) throws
+            AccountLockException {
+
+        AccountUtil.publishEvent(accountLockedEventName, AccountUtil.cloneMap(map));
+    }
+
+    private void publishPostAccountLockedEvent(String accountLockedEventName, Map<String, Object> map, boolean
+            isLockPropertySuccessfullyModified) throws AccountLockException {
+
+        Map<String, Object> eventProperties = AccountUtil.cloneMap(map);
+        eventProperties.put(IdentityEventConstants.EventProperty.UPDATED_LOCKED_STATUS,
```

Comment:
If `eventProperties` map is null, it will cause a NPE. Let's add a `MapUtils.isNotEmpty` to avoid this.